weblab-identity-auth0/src/main/java/ucles/weblab/common/identity/auth0/UserAttributesResolver.java

```diff
@@ -34,18 +34,28 @@ public String getUserId() {
 
         if (authentication instanceof OAuth2Authentication) {
             OAuth2Authentication auth2Authentication = (OAuth2Authentication) authentication;
-            return auth2Authentication.getUserAuthentication().getName();
+            Object principal = auth2Authentication.getUserAuthentication().getPrincipal();
+
+            if (principal instanceof ExtendedUser) {
+                return getUserIdFromExtendedUser((ExtendedUser) principal);
+            } else {
+                return principal.toString();
```

Comment:
@zhacpalomares I don't think this would return the user id, I think principal.toString() will return the username.